subscription-engine/kafka-ws/app/Redis/RedisExpireMgmt.js

```diff
@@ -20,82 +20,150 @@ class RedisExpireMgmt {
     this.expirationMap = new Map();
 
     // TODO:  ADD TLS and  PASSWORD options for Redis
-    // TODO: Implement "retry_strategy"
     // const fs = require('fs');
     // const tls_options = {
     //   ca: [ fs.readFileSync(resolve(__dirname, "./server.crt")) ]
     //   ...
     // };
+    this.nameServicePub = 'redis-expire-pub';
+    this.nameServiceSub = 'redis-expire-sub';
+
+    // redis strategy
+    this.config.retry_strategy = this.retryStrategy.bind(this);
 
     this.clients = {
       pub: redis.createClient(this.config),
       sub: redis.createClient(this.config),
     };
 
+    this.initSubscribe();
```

Comment:
I dont see any gain, because in KafkaTopicsCallbacksMgmt there is an init method because it needs to call some async functions. That’s why it’s not in the builder. 

Do you think it would be more organized? I would then call the builder. What do you think?

Reply:
I have no problem with leaving it as it is, your answer seemed very coherent :smile: